internal/data/notificationsintegrations/notificationsintegrations.go

```diff
@@ -49,6 +49,14 @@ func (r *notificationsintegrationsRepo) Save(ctx context.Context, model *biz.Not
 			return nil, err
 		}
 	}
+
+	if r.Authz != nil {
+		_, err := r.Authz.SetWorkspace(ctx, model.ID, model.Metadata.Workspace, "notifications", "integration")
```

Comment:
Which ID is this? I think this may be wrong.

We need this to be the _reporter's_ ID–as in, Notification's ID for the Integration (or hbi's, or acm's, etc).

This is because for queries and access control, these systems will be using their own IDs, not Kessel's IDs. So for initial use cases to work especially, these have to be reporter IDs for the resource.